lib/kernel/subgroups.c

```diff
@@ -135,6 +135,16 @@ SUB_GROUP_SHUFFLE_T (int)
 SUB_GROUP_SHUFFLE_T (uint)
 SUB_GROUP_SHUFFLE_T (long)
 SUB_GROUP_SHUFFLE_T (ulong)
+#ifdef cl_khr_fp16
+SUB_GROUP_SHUFFLE_T (half)
+/* OpenCL C mangles half 'h' whereas C (clang) mangles if fp16.
```

Comment:
"... whereas C (clang) mangles if fp16." does not make much sense, at least to me.